src/orange/OrangeParams.hh

```diff
@@ -0,0 +1,121 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file OrangeParams.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include "base/CollectionMirror.hh"
+#include "base/Types.hh"
+#include "geometry/BoundingBox.hh"
+#include "Data.hh"
+#include "Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Persistent model data for an ORANGE geometry.
+ *
+ * This class initializes and manages the data used by ORANGE (surfaces,
+ * volumes) and provides a host-based interface for them.
+ */
+class OrangeParams
+{
+  public:
+    //!@{
+    //! References to constructed data
+    using HostRef
+        = OrangeParamsData<Ownership::const_reference, MemSpace::host>;
+    using DeviceRef
+        = OrangeParamsData<Ownership::const_reference, MemSpace::device>;
+    //!@}
+
+    struct Input
+    {
+        using Surfaces = SurfaceData<Ownership::value, MemSpace::host>;
+        using Volumes  = VolumeData<Ownership::value, MemSpace::host>;
+        using VecStr   = std::vector<std::string>;
+
+        Surfaces    surfaces;       //!< Surface definitions
+        Volumes     volumes;        //!< Volume definitions
+        VecStr      surface_labels; //!< Surface names (metadata)
+        VecStr      volume_labels;  //!< Volume names (metadata)
+        BoundingBox bbox;           //!< Outer bounding box (metadata)
+    };
+
+  public:
+    // Construct from a JSON file (if JSON is enabled)
+    explicit OrangeParams(const std::string& json_filename);
+
+    // ADVANCED usage: construct from explicit host data
+    explicit OrangeParams(Input input);
+
+    //// VOLUMES ////
+
+    //! Number of volumes
+    VolumeId::size_type num_volumes() const { return vol_labels_.size(); }
+
+    // Get the label for a placed volume ID
+    const std::string& id_to_label(VolumeId vol_id) const;
+
+    // DEPRECATED alias for find_volume
+    inline VolumeId label_to_id(const std::string& label) const;
```

Comment:
Could this just be removed? I noticed it doesn't show up in the tests

Reply:
I could change the VecGeom interface or remove this, your call

Reply:
I'd say let's change the VecGeom interface to use `find_volume` and then just remove this

Reply:
I think I was worried about changing the VecGeom name because that might lead me through a cascade of renaming and tweaking names 😂 but this change is quite simple and localized.